receiver/redisreceiver/pdata_test.go

```diff
@@ -0,0 +1,204 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package redisreceiver
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/consumer/pdata"
+)
+
+func TestServiceName(t *testing.T) {
+	const serviceName = "foo-service"
+	rm := testGetMetricData(t, usedMemory(), serviceName)
+	const key = "service.name"
+	attrVal, ok := rm.Resource().Attributes().Get(key)
+	assert.True(t, ok)
+	assert.Equal(t, serviceName, attrVal.StringVal())
+}
+
+func TestMemoryMetric(t *testing.T) {
+	rm := testGetMetricData(t, usedMemory(), "")
+
+	const metricName = "redis/memory/used"
+	const desc = "Total number of bytes allocated by Redis using its allocator"
+	const units = "By"
+	const ptVal = 854160
+
+	ilms := rm.InstrumentationLibraryMetrics()
+	assert.Equal(t, 1, ilms.Len())
+	assert.Equal(t, 2, rm.Resource().Attributes().Len())
+	ilm := ilms.At(0)
+	ms := ilm.Metrics()
+	m := ms.At(0)
+	assert.Equal(t, metricName, m.Name())
+	assert.Equal(t, desc, m.Description())
+	assert.Equal(t, units, m.Unit())
+	assert.Equal(t, pdata.MetricDataTypeIntGauge, m.DataType())
+	assert.Equal(t, int64(ptVal), m.IntGauge().DataPoints().At(0).Value())
+}
+
+func TestUptimeInSeconds(t *testing.T) {
+	pdm := testGetMetric(t, uptimeInSeconds(), "")
+	const units = "s"
+	v := 104946
+
+	assert.Equal(t, units, pdm.Unit())
+	assert.Equal(t, int64(v), pdm.IntSum().DataPoints().At(0).Value())
+}
+
+func TestUsedCpuSys(t *testing.T) {
+	pdm := testGetMetricData(t, usedCPUSys(), "")
+	const units = "s"
+	v := 185.649184
+
+	m := pdm.InstrumentationLibraryMetrics().At(0).Metrics().At(0)
+	assert.Equal(
+		t,
+		pdata.MetricDataTypeDoubleSum,
+		m.DataType(),
+	)
+	assert.Equal(t, units, m.Unit())
+	assert.Equal(t, v, m.DoubleSum().DataPoints().At(0).Value())
+}
+
+func TestMissingMetricValue(t *testing.T) {
+	redisMetrics := []*redisMetric{{key: "config_file"}}
+	_, warnings, err := testFetchMetrics(redisMetrics)
+	require.Nil(t, err)
+	// treat a missing value as not worthy of a warning
+	require.Nil(t, warnings)
+}
+
+func TestMissingMetric(t *testing.T) {
+	redisMetrics := []*redisMetric{{key: "foo"}}
```

Comment:
It's not immediately clear how this is different than the previous case or why the warning results would be different.  Could be helpful to assert on error/warning content to help provide more info.

Reply:
Yeah, that's true. I've added a comment to clarify.